### PR TITLE
Creating SQL tables

### LINKED ISSUE #13: Create SQL statements for creating DB entities
1. create a database
2. create separate tables, indexes for all the entities and their relationships.

(If you can name the sql files like

001-create-volunteers.sql
002-<whatever_you_do_now>.sql

and so on - people will know in which order they should execute those files.)

### PR DESCRIPTION
**Pull request by mykyta-protsenko**

### CODE REVIEW

File: sql/001_create_small_business_table.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE `small_businesses` (
+  `id` int NOT NULL,
+  `name` varchar(45) NOT NULL,
+  `city` varchar(45) NOT NULL,
+  `zip` int NOT NULL,
+  `cause area` varchar(45) NOT NULL,
```
Comment (mykyta-protsenko) [inline sql/001_create_small_business_table.sql:6]: I suggest being more consistent in field naming - having a space in a field name is technically acceptable, but in practice it creates issues for fellow developers - they have to remember that this field have to be escaped.

Please rename it to `cause_area`

File: sql/002_create_volunteers_table.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE `volunteers` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `first_name` varchar(45) NOT NULL,
+  `last_name` varchar(45) NOT NULL,
+  `points` int NOT NULL,
+  `contact` varchar(45) NOT NULL,
+  `perk_date` varchar(45) DEFAULT NULL,
```
Comment (mykyta-protsenko) [inline sql/002_create_volunteers_table.sql:7]: Can you please explain how a perk_date is related to volunteers?

File: sql/003_create_nonprofits_table.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE `non_profits` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `name` varchar(45) NOT NULL,
+  `city` varchar(45) NOT NULL,
+  `zipcode` int NOT NULL,
+  `points` int NOT NULL,
+  `total_volunteers` int NOT NULL,
+  `total_hours` int NOT NULL,
+  `approved_hours` int NOT NULL,
```
Comment (mykyta-protsenko) [inline sql/003_create_nonprofits_table.sql:9]: are we going to record approved and pending hours as a running total or per any given day?

File: sql/004.sql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE `perks` (
+  `id` int NOT NULL AUTO_INCREMENT,
```
Comment (mykyta-protsenko) [inline sql/004.sql]: How do we find out which business perks are related to?

File: sql/005_create_business_posting_table.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE `business_posting` (
+  `business_name` varchar(45) NOT NULL,
```
Comment (mykyta-protsenko) [inline sql/005_create_business_posting_table.sql:2]: `business_name` makes for a bad primary key. Primary key has to be unique. It is totally fine to have two business /posting for two separate 7/11 stores.

That is going to create a conflict.

File: sql/009_join_business_with_perks.sql [hunk 6]
```diff
@@ -0,0 +1,12 @@
+SELECT 
+sb.name business_name,
+ p.name AS perk_given, 
+ p.status,
+ p.starts,
+ p.ends
+FROM  perks p
+JOIN small_businesses sb
+    ON p.id = sb.id
```
Comment (mykyta-protsenko) [inline sql/009_join_business_with_perks.sql]: Here is a problem

Let say we have perk with ID=1 which is entered by business A (the business A's ID=2).

We are never going to see this perk in this query. You need to use a foreign key instead. I sent you an example of a foreign key.

On top of that - we don't need `select` statements to be recorded alongside with DDL statements - but we can address it later.

File: sql/004.sql [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  `name` varchar(60) NOT NULL,
+  `point_price` int NOT NULL,
+  `starts` date NOT NULL,
+  `ends` date NOT NULL,
+  `combination with other offers` tinyint NOT NULL,
+  `perk photo` varchar(100) NOT NULL,
```
Comment (mykyta-protsenko) [inline sql/004.sql]: Have you got a chance to sync with @anavianm and @ankoanju ? Are they going to store a URL here? If yes, then 100 characters may not be enough. And you will need a totally different field if they need to store binary data.

**Review by baffour1999 — APPROVED** [review]

File: sql/003_create_nonprofits_table.sql (hunk 3 above)
Comment (baffour1999) [reply sql/003_create_nonprofits_table.sql:9]: kenthia says it is a running total

**Pull request merged**